### PR TITLE
[Modules] Adds defaultHostname output to Microsoft.Web/sites and Microsoft.Web/staticSites modules

### PR DESCRIPTION
**Pull request by DanielLarsenNZ**
# Description

Adds `defaultHostname` output to `Microsoft.Web/sites` and `Microsoft.Web/staticSites` modules. This output is required for Bicep templates that deploy Azure Front Door or App Gateway with App Services, Functions and Static Websites.

# Type of Change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] Update to documentation

# Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (readme)
- [X] I did format my code

### CODE REVIEW

**Comment by DanielLarsenNZ** [comment]
Fixes #1788

File: modules/Microsoft.Web/sites/readme.md [hunk 1]
```diff
@@ -390,6 +390,7 @@ userAssignedIdentities: {
 | `resourceGroupName` | string | The resource group the site was deployed into. |
 | `resourceId` | string | The resource ID of the site. |
 | `systemAssignedPrincipalId` | string | The principal ID of the system assigned identity. |
+| `defaultHostname` | string | Default hostname of the app. |
```
Comment (ahmadabdalla) [inline modules/Microsoft.Web/sites/readme.md:393]: Please run the 'Set-ModuleReadMe.ps1' script to regenerate the documentation after the changes.
image

The pipeline static code check fails as it detects a difference between the documentation submitted vs. the document that gets autogenerated during the test for comparison:

This is the error when I attempt to replay your branch changes
image
Here is a guide on how to use the [Set-ModuleReadMe.ps1](https://github.com/Azure/ResourceModules/wiki/Contribution%20guide%20-%20Generate%20module%20Readme) script

File: modules/Microsoft.Web/staticSites/readme.md [hunk 2]
```diff
@@ -269,6 +269,7 @@ userAssignedIdentities: {
 | `resourceGroupName` | string | The resource group the static site was deployed into. |
 | `resourceId` | string | The resource ID of the static site. |
 | `systemAssignedPrincipalId` | string | The principal ID of the system assigned identity. |
+| `defaultHostname` | string | The default autogenerated hostname for the static site. |
```
Comment (ahmadabdalla) [inline modules/Microsoft.Web/staticSites/readme.md:272]: Please see other comment as there is a similar error

image

**Review by ahmadabdalla — CHANGES_REQUESTED** [review]
Thank you for your contribution :) I provided some comments regarding the documentation

File: modules/Microsoft.Web/sites/readme.md (hunk 1 above)
Comment (DanielLarsenNZ) [reply modules/Microsoft.Web/sites/readme.md:393]: OK thanks Ahmad, will do 👍

Comment (ahmadabdalla) [reply modules/Microsoft.Web/sites/readme.md]: @DanielLarsenNZ have you tested the deployment with the new changes?

Comment (DanielLarsenNZ) [reply modules/Microsoft.Web/sites/readme.md]: Yes. WOMM. I have an Azure Quickstart template in progress that uses this (patched) module.

**Review by ahmadabdalla — APPROVED** [review]

**Pull request merged**